menpofit/dpm/cpp/qp_one_sparse.h

```diff
@@ -0,0 +1,151 @@
+#include <math.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <iostream>
+using namespace std;
+
+#define MAX(A,B) ((A) < (B) ? (B) : (A))
+#define MIN(A,B) ((A) > (B) ? (B) : (A))
+#define INDEX(X,I) (*(int *)(X + I*mm + m))
+int n, m, mm;
```

Comment:
I would much rather these were passed as arguments - presumably they are global due to the use `comp` below, however, if you make the function pointer inline I think it will actually 'catpure' `m` anyway.